tests/regression/test_linesmoother.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+from firedrake import *
+
+
+@pytest.fixture(params=[1, 2, 3],
+                ids=["Extruded Interval", "Extruded Triangle Mesh", "Extruded Quadrilateral Mesh"])
+def mesh(request):
+    layers = 5
+    layer_height = 1/layers
+    if request.param == 1:
+        m = IntervalMesh(10, 100)
+        S1family = "CG"
+        S2family = "DG"
+        degree_shift = 1
+        cell = interval
+    if request.param == 2:
+        m = RectangleMesh(10, 10, 100, 100)
+        S1family = "BDM"
+        S2family = "DG"
+        degree_shift = 1
+        cell = triangle
+    if request.param == 3:
+        m = RectangleMesh(10, 10, 100, 100, quadrilateral=True)
+        S1family = "RTCF"
+        S2family = "DG"
+        degree_shift = 1
+        cell = quadrilateral
+    mex = ExtrudedMesh(m, layers, layer_height)
+    return {'mesh': mex, 'layers': layers, 'S1family': S1family, 'S2family': S2family,
+            'degree_shift': degree_shift, 'cell': cell}
+
+
+@pytest.fixture
+def expected(mesh):
+    if mesh["S1family"] == "CG":
+        return [8, 13]
+    elif mesh["S1family"] == "BDM":
+        return [10, 26]
+    elif mesh["S1family"] == "RTCF":
+        return [9, 20]
+
+
+def test_linesmoother(mesh, expected):
+
+    nits = []
+    for degree in range(2):
+        S1 = FiniteElement(mesh["S1family"], mesh["cell"], degree+mesh["degree_shift"])
+        S2 = FiniteElement(mesh["S2family"], mesh["cell"], degree)
```

Comment:
Cell, available from `mesh.ufl_cell().sub_cells()[0]`, `degree_shift` is always 1.

Reply:
done, but superceded by change below